client/actions/assignments/ui.js

```diff
@@ -30,6 +30,26 @@ const loadAssignments = (
     return dispatch(queryAndSetAssignmentListGroups(filterByState));
 };
 
+const myAssignments = (
+    filterByState = null
+) => (dispatch) => dispatch(queryAndGetMyAssignments(filterByState));
+
+const queryAndGetMyAssignments = (filterByState) => (
+    (dispatch, getState) => {
+        let querySearchSettings = selectors.getAssignmentSearch(getState());
```

Comment:
Whenever we modify the response from a call to `selector.` you must cloneDeep the object returned. Otherwise any modifications to the object will be reflected on other calls to the `selector.` function